packages/outstatic/src/pages/edit-document/index.tsx

```diff
@@ -70,33 +74,130 @@ export default function EditDocument({ collection }: EditDocumentProps) {
     skip: slug === 'new' || !slug
   })
 
+  const { data: metadata } = useDocumentQuery({
+    variables: {
+      owner: repoOwner || session?.user?.login || '',
+      name: repoSlug,
+      filePath: `${repoBranch}:${
+        monorepoPath ? monorepoPath + '/' : ''
+      }${contentPath}/metadata.json`
+    },
+    fetchPolicy: 'network-only'
+  })
+
   const onSubmit = async (data: Document) => {
     setLoading(true)
     try {
       const document = methods.getValues()
-      const content = mergeMdMeta({ ...data })
+      let content = mergeMdMeta({ ...data })
+      const { data: matterData } = matter(content)
       const oid = await fetchOid()
       const owner = repoOwner || session?.user?.login || ''
       const newSlug = document.slug
 
+      const fullContentPath = `${contentPath}/${collection}/${slug}.md`
+
       // If the slug has changed, commit should delete old file
       const oldSlug = slug !== newSlug && slug !== 'new' ? slug : undefined
 
-      const commitInput = createCommitInput({
+      const capi = createCommitApi({
+        message: oldSlug
+          ? `chore: Updates ${newSlug} formerly ${oldSlug}`
+          : `chore: Updates/Creates ${newSlug}`,
         owner,
-        slug: newSlug,
-        oldSlug,
-        content,
-        oid,
-        files,
-        repoSlug,
-        repoBranch,
-        contentPath,
-        monorepoPath,
-        collection
+        oid: oid ?? '',
+        name: repoSlug,
+        branch: repoBranch
       })
 
-      await createCommit({ variables: commitInput })
+      if (oldSlug) {
+        capi.removeFile(
+          `${
+            monorepoPath ? monorepoPath + '/' : ''
+          }${contentPath}/${collection}/${oldSlug}.md`
+        )
+      }
+
+      if (files.length > 0) {
+        files.forEach(({ filename, blob, type, content: fileContents }) => {
+          // check if blob is still in the document before adding file to the commit
+          if (blob && content.search(blob) !== -1) {
+            const randString = window
+              .btoa(Math.random().toString())
+              .substring(10, 6)
+            const newFilename = filename
+              .toLowerCase()
+              .replace(/[^a-zA-Z0-9-_\.]/g, '-')
+              .replace(/(\.[^\.]*)?$/, `-${randString}$1`)
+
+            const filePath = (() => {
+              switch (type) {
+                case 'images':
+                  return IMAGES_PATH
+                default:
+                  assertUnreachable(type)
+              }
+            })()
+
+            capi.replaceFile(
+              `${
+                monorepoPath ? monorepoPath + '/' : ''
+              }public/${filePath}${newFilename}`,
+              fileContents,
+              false
+            )
+
+            // replace blob in content with path
+            content = content.replace(blob, `/${filePath}${newFilename}`)
+          }
+        })
+      }
+
+      capi.replaceFile(fullContentPath, content)
```

Comment:
Here we should use the full path with the `newSlug` const.

```suggestion
      capi.replaceFile(
        `${
          monorepoPath ? monorepoPath + '/' : ''
        }${contentPath}/${collection}/${newSlug}.md`,
        content
      )
```